resolver-dns/src/main/java/io/netty/resolver/dns/ShuffledDnsServerAddressStream.java

```diff
@@ -27,11 +27,16 @@
     private int i;
 
     ShuffledDnsServerAddressStream(InetSocketAddress[] addresses) {
-        this.addresses = addresses.clone();
+        this.addresses = addresses;
```

Comment:
add a comment that the passed in InetSocketAddress[] is not cloned and so the caller must ensure the same array is not modified later on ?